dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileBig.cpp

```diff
@@ -138,15 +138,24 @@ size_t ColumnFileBigReader::readRowsRepeatedly(MutableColumns & output_cols, siz
         const auto & columns = cached_pk_ver_columns.at(block_index);
         const auto & pk_column = columns[0];
 
-        actual_read += copyColumnsData(columns, pk_column, output_cols, rows_start_in_block, rows_in_block_limit, range);
+        auto [offset, rows] = copyColumnsData(columns, pk_column, output_cols, rows_start_in_block, rows_in_block_limit, range);
+        // For DMFile, records are sorted by row key. Only the first block and the last block will be filtered by range.
```

Comment:
`p0 p1, p2, p3`
There are possibilities that `range` can filter out `p0` and part of `p1`. Although it won't happen by now, I am afraid it could be in the future after somebody changes the code.
I mean we better don't rely on the assumption that `range` can only filter out part of `p0` or `p3`, it is dangerous.

Reply:
like
`p0: [0, 1000), p1:[1000, 2000), p2:[2000, 3000), p3:[3000, 4000)`
`range: [0, 1003)`
?